src/main/kotlin/com/nikitahohulia/listeningplatform/repository/CustomPostRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.nikitahohulia.listeningplatform.repository
+
+import com.nikitahohulia.listeningplatform.entity.Post
+import org.bson.types.ObjectId
+
+interface CustomPostRepository {
+
+    fun getPostById(id: ObjectId): Post?
```

Comment:
```suggestion
    fun findPostById(id: ObjectId): Post?
```

since you return null if nothing found